### DISCUSSION TITLE
Question about the version of skywalking-goapi

### DISCUSSION THREAD

**Discussion by LinuxSuRen** [apache/skywalking#11476] [General]
I didn't find a git tag from [apache/skywalking-goapi](https://github.com/apache/skywalking-goapi). There is only one branch exist. Below are some downstream project of it:

* apache/skywalking-go@0.3.0 is using v0.0.0-20230314034821-0c5a44bb767a
* opentelemetry-collector-contrib is using v0.0.0-20220121092418-9c455d0dda3f

It would be more clear if there is a semantic version.

**Comment by wu-sheng** [comment]
We don't have a version for that. goapi is just a generated codes of SkyWalking data collect and query protocols. It is a combination.

**Reply by wu-sheng** [reply]
About OTEL, we don't control them.

**Reply by LinuxSuRen** [reply]
I konw the OTEL is just a regular downstream project. Can I think the skywalking-goapi is just an internal library, so it's not necessary to have a fixed version?

I totally understand all the code of goapi was generated. But it still has the connection with the core project (AKA apache/skywalking). So, it has an implict version which is same with the core project. Please correct me if I mis-understand it.

**Reply by wu-sheng** [reply]
> I konw the OTEL is just a regular downstream project. Can I think the skywalking-goapi is just an internal library, so it's not necessary to have a fixed version?

Because right now, we don't have a protocol release mechanism to match goapi.

The original protocol repositories are 
- https://github.com/apache/skywalking-data-collect-protocol
- https://github.com/apache/skywalking-query-protocol

You can see the goapi repo is just locking in commits from https://github.com/apache/skywalking-goapi/blob/main/dependencies.sh.

So, we don't have a known requirement to release the goapi repo through a version control mechanism. That is the only reason, we can't find a use case.

From user's perspective, usually, they don't need to worry about this. SkyWalking 8.0 -> 9.5 releases keep high protocol capability, especially from the data reporting side. Users just need to choose the latest backend/UI releases and it works.

___

I am wondering what is the case you are asking for this?

**Reply by LinuxSuRen** [reply]
I'm searching the related description or discussion about the goapi version. Your explanation is enough for me. Thanks for that.
